dateparser/date_parser.py

```diff
@@ -32,10 +34,65 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, prefer_future=None, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        timestr = timestr if not isinstance(timestr, str) else StringIO(timestr)
+
+        # Set right prefer_future var
+        prefer_future = prefer_future if prefer_future is not None \
+            else settings.PREFER_DATES_FROM_FUTURE
+
+        # Parse timestr
+        res = self._parse(timestr, **kwargs)
+
+        if res is None:
+            raise ValueError("unknown string format")
+
+        # Fill in missing date
+        given_fields = []  # fields set by
+        new_date = default
+
+        for field in ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']:
```

Comment:
The given_fields is used for a different purpose. While, populating data it saves the fields there which should not be overridden. If you think the initial implementation with the repl = {} variable is more efficient, I'm also fine with that.